### PR TITLE
Add validation script for Image Classification pipeline

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR will add a validation script for image classification pipelines. The goal is to make testing easier and make sure the pipelines work as intended.

This also updates IC pipelines to work with `numpy.ndarray` out of the box

Note: In it's current form it needs the dataset to be pre-downloaded on local filesystem.

Usage:
```bash
python validation_script.py --dataset-path ~/data/imagenette-160/val
```
Output:
```bash
DeepSparse Engine, Copyright 2021-present / Neuralmagic, Inc. version: 0.11.2 (ca6e29a5) (release) (optimized) (system=avx512, binary=avx512)
100%|████████████████| 500/500 [00:06<00:00, 77.58it/s, Running Accuracy=99.00%]
Accuracy: 99.00 %
```

Output from `python validation_script.py --help`

```bash
Usage: validation_script.py [OPTIONS]

  Validation Script for Image Classification Models

Options:
  --dataset-path, --dataset_path DIRECTORY
                                  Path to the validation dataset  [required]
  --model-path, --model_path TEXT
                                  Path/SparseZoo stub for the Image
                                  Classification model to be evaluated.
                                  Defaults to resnet50 trained on
                                  Imagenette  [default: zoo:cv/classification/
                                  resnet_v1-50/pytorch/sparseml/imagenette/
                                  base-none]
  --batch-size, --batch_size INTEGER
                                  Test batch size, must divide the dataset
                                  evenly  [default: 1]
  --help    
```

### CODE REVIEW

File: src/deepsparse/image_classification/validation_script.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    "evaluated. Defaults to resnet50 trained on Imagenette",
+    show_default=True,
+)
+@click.option(
+    "--batch-size",
+    "--batch_size",
+    type=int,
+    default=1,
+    show_default=True,
+    help="Test batch size, must divide the dataset evenly",
```
Comment (dbogunowicz) [inline src/deepsparse/image_classification/validation_script.py:93]: It does not have to divide the dataset evenly if you apply a simple trick ("Machine learning engineers hate him for this ONE SIMPLE TRICK").

data_loader = DataLoader(training_dataset, batch_size=batch_size, shuffle=True, **drop_last=True**)

Comment (dbogunowicz) [reply src/deepsparse/image_classification/validation_script.py:93]: But this is obviously only good if you are running verification on the test set; NOT if you want to run inference in prod :)

File: src/deepsparse/image_classification/validation_script.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            [
+                transforms.ToTensor(),
+                transforms.Resize(size=(224, 224)),
+            ]
+        ),
+    )
+
+    data_loader = DataLoader(
+        dataset=dataset,
+        batch_size=batch_size,
```
Comment (dbogunowicz) [inline src/deepsparse/image_classification/validation_script.py:112]:
```suggestion
        batch_size=batch_size,
        drop_last=True,
```

**Review by dbogunowicz — COMMENTED** [review]
@rahul-tuli you've been keeping me busy a lot lately. Good and quality PRs.

File: src/deepsparse/image_classification/validation_script.py (hunk 1 above)
Comment (rahul-tuli) [reply src/deepsparse/image_classification/validation_script.py:93]: Thank you for the fix applied this in upcoming commit

**Review by dbogunowicz — APPROVED** [review]
let it snow

**Review by bfineran — APPROVED** [review]
approving and will land to `ic-pipelines` branch. noting that fixes will be added there for serialization and image size override
